Bacpac/Restore-BcDatabaseFromArtifacts.ps1

```diff
@@ -98,14 +101,17 @@ try {
             if ($multitenant) {
                 $dbName = "$($databasePrefix)tenant"
             }
-            Import-Module sqlps -ErrorAction SilentlyContinue
-            $sqlpsModule = get-module sqlps
-            if (-not $sqlpsModule) {
-                import-module SqlServer
-                $SqlModule = get-module SqlServer
-                if (-not $SqlModule) {
-                    throw "You need to have a local installation of SQL or you need the SqlServer PowerShell module installed"
-                }
+            if($sqlModuleToUse -eq "sqlps") {
+                Import-Module sqlps
+                $SqlModule = Get-Module sqlps
+                Write-Host "Using sqlps module."
+            } elseif ($sqlModuleToUse -eq "sqlserver") {
```

Comment:
Current functionality will try sqlps and if that is not found - then it will try sqlserver (which will work in PS7 where SQLPS is not available). The new functionality will break this behavior and make the function always fail on PS7

A better (and non-breaking) fix would be to replace the sqlModuleToUse with a switch parameter called UseSqlServerModule - and then not try sqlps - which I guess is what you want?

Reply:
And then maybe create a setting called UseSqlServerModule - which you use as default value for the parameter - meaning that people can set this setting on the machine to always use SqlServer

Reply:
Will do.

Yes. To be more specific about my problem, sqlps always fails for for me, while sqlserver always succeeds. This has held true for me on different machines and different versions of PowerShell for several months. If I uninstall sqlps, it just comes back when I run the existing BcContainerHelper scripts and never even tries to use sqlserver. At this point, I just want to make the choice for myself.